app/src/main/java/com/example/dungeontest/EditorScreen.kt

```diff
@@ -33,76 +41,141 @@ import androidx.navigation.NavController
-            )
-            MultiFloatingActionButton(items)
-        },
-        floatingActionButtonPosition = FabPosition.End
-    ) { innerPadding ->
-        Box(
-            modifier = Modifier
-                .fillMaxSize()
-                .padding(innerPadding)
-        ) {
+            }
+        }
+    }
 
-            Column(
-                verticalArrangement = Arrangement.spacedBy(16.dp),
-                horizontalAlignment = Alignment.CenterHorizontally,
-                modifier = Modifier.fillMaxSize()
+
+    if (showLoadingOverlay.value) {
+        LoadingOverlay(
+            message = statusMessage.value,
+        )
+    } else {
+        Scaffold(
+            snackbarHost = {
+                SnackbarHost(hostState = snackbarHostState)
+            },
+            topBar = {
+                CenterAlignedTopAppBar(
+                    title = { Text("Edit Map") },
+                    navigationIcon = {
+                        IconButton(onClick = {
+                            scope.launch {
+                                drawerState.open()
+                                keyboardController?.hide()
+                                focusManager.clearFocus()
+                            }
+                        }) {
+                            Icon(Icons.Filled.Menu, contentDescription = "Open nav drawer")
+                        }
+                    }
+                )
+            },
+            floatingActionButton = {
+                // for Scott
+                val items = listOf(
+                    MiniFabItems(Icons.Filled.Check, "Save", onClick = {
+                        //onClick logic goes here
+                    }),
+                    MiniFabItems(Icons.Filled.Edit, "Rename Node", onClick =
+                    {
+                        //onClick logic goes here
+                    }),
+                    MiniFabItems(Icons.Filled.Add, "Add Edge", onClick = {
+                        //onClick logic goes here
+                    }),
+                    MiniFabItems(Icons.Filled.Clear, "Delete Edge", onClick = {
+                        //onClick logic goes here
+                    }),
+                )
+                MultiFloatingActionButton(items)
+            },
+            floatingActionButtonPosition = FabPosition.End
+        ) { innerPadding ->
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(innerPadding)
             ) {
-                MapVisualization(testDot)
+                Column(
+                    verticalArrangement = Arrangement.spacedBy(16.dp),
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    modifier = Modifier.fillMaxSize()
+                ) {
+                    if(mapViewModel.transitoryMapRecord!!.dotString != null){
+                        MapVisualization(mapViewModel.transitoryMapRecord!!.dotString!!)
+                    }
```

Comment:
I think the previous 3 lines could be made slightly simpler/more-concise by extracting something to a local variable (which also lets you avoid the non-null assertion on the thing which you just null checked, because this simpler form allows the kotlin compiler to realize that, inside the if block, the variable that you just tested for nullity won't be null.
```
val pendingMapDotString = mapViewModel.transitoryMapRecord!!.dotString 
                    if(pendingMapDotString != null){
                        MapVisualization(pendingMapDotString)
                    }
```
Another alternative (suggested by ide) is this:
```
mapViewModel.transitoryMapRecord!!.dotString?.let { 
                        MapVisualization(dotAsString = it)
                    }
```
@pepsipotty @EricMiddie  what do you think?

Reply:
I like the second suggestion better. The first suggestion addresses readability but makes it a lil less concise while the second improves upon readability whilst also trimming away excess fat

Reply:
Addressed